contracts/multisig-prover/src/types.rs

```diff
@@ -40,3 +82,54 @@ pub struct Operator {
     pub weight: Uint256,
     pub signature: Option<Signature>,
 }
+
+#[cw_serde]
+pub struct ProofID(HexBinary);
+
+impl Display for ProofID {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0.to_hex())
+    }
+}
+
+impl<'a> PrimaryKey<'a> for ProofID {
+    type Prefix = ();
+    type SubPrefix = ();
+    type Suffix = ProofID;
+    type SuperSuffix = ProofID;
+
+    fn key(&self) -> Vec<Key> {
+        vec![Key::Ref(self.0.as_slice())]
+    }
+}
+
+impl KeyDeserialize for ProofID {
+    type Output = ProofID;
+
+    fn from_vec(value: Vec<u8>) -> StdResult<Self::Output> {
+        Ok(from_binary(&value.into()).expect("violated invariant: ProofID is not deserializable"))
+    }
+}
+
+impl From<HexBinary> for ProofID {
+    fn from(id: HexBinary) -> Self {
+        Self(id)
+    }
+}
+
+impl From<&[u8]> for ProofID {
+    fn from(id: &[u8]) -> Self {
+        Self(id.into())
+    }
+}
+
+impl ProofID {
+    pub fn new(batch_id: &BatchID, session_id: &Uint64) -> Self {
+        let mut hasher = Keccak256::new();
+
+        hasher.update(Into::<&[u8]>::into(batch_id));
```

Comment:
Do we need to separator between batch_id and session_id? Asking cuz I'm not sure what exactly does `hasher.update` do.